### DISCUSSION TITLE
Template Flags

### DISCUSSION THREAD

**Discussion by pcdiem** [arendst/Tasmota#13342] [Ideas]
Are there any plans to remove or re-use the template flags? The PWM Dimmer module provides support for 1, 2 and 3 button switches and dimmers. It was originally written to support the Martin-Jerry SD01 dimmers and clones. These dimmer have 5 LED's to indicate the current dimmer level and the lowest one is tied to the relay so it's always on when the relay is on. The logic to determine how many LED's to turn on makes the assumption that one LED will always be on. THE PWM Dimmer can be used with most other switches/dimmers and most of them do not have an LED that is tied to the relay. I need a way to define whether there is an LED tied to the relay or not. The most logical place to define this would be in a flag the template.

**Comment by barbudor** [comment]
What about always driving 5 leds from Tasmota (Led1 to Led5) and assigning Led1 to an unused GPIO on devices where the led is tighted to the relay ?

Personaly I would find FLAGS (if like hex codes) in the template cryptic. If we should have a flag for that, I would prefer to have a SetOption and use the recently added optional "CMND" field in the template to explicitly enable the SO.

**Comment by arendst** [comment]
As we currently do not use the flag field I prefer not using it at all. The above solution sounds bettter to me or just providing a different template for the devices that need it.

**Comment by pcdiem** [comment]
@barbudor 's solution would be a good one except Tasmota only allows four LED's.

**Reply by barbudor** [reply]
🤦‍♂️
what a moron Barbudor is
But may be Theo was talking about the SetOption ?

Otherwise we need more Leds, more PWMS, more counters more ... and more of everything 🤣
